R/gppt_graph.R

```diff
@@ -0,0 +1,266 @@
+## Detailed Level GPPT Graphs
+
+# install.packages("devtools")
+# install.packages("tidyverse")
+# devtools::install_github("NHS-South-Central-and-West/scwplot")
+# devtools::install_github("bbc/bbplot")
+
+
+library(tidyverse)
+library(bbplot)
+library(scwplot)
+
+#### Practice Level Graphs
+## practice_code and question_number to be dependent on the selected item on the app
+gppt_practice_graph <- filter(
+  gppt_final, practice_code == "K84028",
+  question_number == "Q21", summary_flag == FALSE,
+  total_flag == FALSE
+)
+
+
+# gppt_practice_graph <- gppt_practice_graph[order(gppt_practice_graph$sort_order),]
+
+
+practice_stack <- ggplot(gppt_practice_graph, aes(x = year, y = value, fill = reorder(answers, sort_order))) +
+  geom_bar(
+    stat = "identity",
+    position = "stack"
+  ) +
+  geom_hline(yintercept = 0, size = 1, colour = "#333333") +
+  theme_scw() +
+  scale_fill_diverging(discrete = TRUE) +
+  scale_y_continuous(labels = scales::label_number_si()) +
+  labs(
+    title = gppt_practice_graph$question,
+    subtitle = gppt_practice_graph$practice_name,
+    y = "Number of Patients",
+    x = "Year"
+  )
+practice_stack
+
+
+
+practice_perc <- ggplot(gppt_practice_graph, aes(x = year, y = value, fill = reorder(answers, sort_order))) +
+  geom_bar(
+    stat = "identity",
+    position = "fill"
+  ) +
+  theme_scw() +
+  geom_hline(yintercept = 0, size = 1, colour = "#333333") +
+  scale_y_continuous(
+    name = "Percentage of Patients",
+    labels = scales::label_percent()
+  ) +
+  scale_fill_diverging(discrete = TRUE) +
+  labs(
+    title = gppt_practice_graph$question,
+    subtitle = gppt_practice_graph$practice_name,
+    x = "Year"
+  )
+practice_perc
+
+
+#### PCN Level Graphs
+## pcn_code and question_number to be dependent on the selected item on the app
+gppt_pcn_graph <- gppt_final %>%
```

Comment:
I'd replace all magrittr pipes (%>%) with the base R pipe (|>), to reduce dependencies.